tests/test_envfile_optional.py

```diff
@@ -0,0 +1,61 @@
+import os
+import shutil
+from pathlib import Path
+
+
+_STUB_PATH = Path(__file__).parent / "stubs"
+
+
+def _stub_env(extra: dict[str, str] | None = None) -> dict[str, str]:
+    pythonpath = os.pathsep.join(
+        filter(None, [str(_STUB_PATH), os.environ.get("PYTHONPATH")])
+    )
+    env = {"PYTHONPATH": pythonpath}
+    if extra:
+        env.update(extra)
+    return env
+
+
+def _copy_fixture_project(projects, project_key: str, tmp_path: Path) -> Path:
+    destination = tmp_path / f"{project_key.replace('/', '_')}_copy"
+    if destination.exists():
+        shutil.rmtree(destination)
+    shutil.copytree(projects[project_key], destination)
+    return destination
+
+
+def test_optional_envfile_missing(run_poe_subproc, projects, tmp_path):
+    project_path = _copy_fixture_project(projects, "envfile_optional", tmp_path)
+
+    optional_env = project_path / ".env"
+    if optional_env.exists():
+        optional_env.unlink()
+
+    result = run_poe_subproc(
+        "print-var",
```

Comment:
if you do this (or set `NO_COLOR` in the env) then there's no need to stub Pastel.
```suggestion
        "--no-ansi", ""print-var",
```